tests/wpunit/Tribe/Tickets/Ticket_ObjectTest.php

```diff
@@ -0,0 +1,595 @@
+<?php
+namespace Tribe\Tickets;
+
+use Tribe__Tickets__Ticket_Object as Ticket_Object;
+use Tribe__Tickets__Ticket_Object as RSVP;
+use Tribe__Tickets__Tickets as Ticket;
+use Tribe__Tickets__Commerce__PayPal__Main as PayPal;
+use Tribe__Tickets__Data_API as Data_API;
+use Tribe__Tickets__Global_Stock as Global_Stock;
+
+use Tribe\Events\Test\Factories\Event;
+use Tribe\Tickets\Test\Commerce\RSVP\Ticket_Maker as RSVP_Ticket_Maker;
+use Tribe\Tickets\Test\Commerce\PayPal\Ticket_Maker as PayPal_Ticket_Maker;
+use Tribe\Tickets\Test\Commerce\Attendee_Maker as Attendee_Maker;
+
+class Ticket_ObjectTest extends \Codeception\TestCase\WPTestCase {
+	use RSVP_Ticket_Maker;
+	use PayPal_Ticket_Maker;
+	use Attendee_Maker;
+
+	private $timezone = 'America/New_York';
+
+	private $later_date = '';
+
+	private $earlier_date = '';
+
+	public function setUp() {
+		// before
+		parent::setUp();
+
+		// your set up methods here
+		$this->factory()->event = new Event();
+
+		$this->later_date = strtotime( '+5 days' );
+
+		$this->earlier_date = strtotime( '-5 days' );
+
+		// Enable Tribe Commerce.
+		add_filter( 'tribe_tickets_commerce_paypal_is_active', '__return_true' );
+		add_filter( 'tribe_tickets_get_modules', function ( $modules ) {
+			$modules['Tribe__Tickets__Commerce__PayPal__Main'] = tribe( 'tickets.commerce.paypal' )->plugin_name;
+
+			return $modules;
+		} );
+
+		// Reset Data_API object so it sees Tribe Commerce.
+		tribe_singleton( 'tickets.data_api', new Data_API );
+	}
+
+	public function tearDown() {
+		// your tear down methods here
+
+		// then
+		parent::tearDown();
+	}
+
+	/**
+	 * @return Ticket_Object
+	 */
+	private function make_instance() {
+		return new Ticket_Object();
+	}
+
+	/**
+	 * Wrapper function: Get ticket object from ticket ID
+	 *
+	 * @param int $ticket_id
+	 * @return Ticket_Object
+	 */
+	public function get_ticket( $event_id, $ticket_id ) {
+		$provider = tribe_tickets_get_ticket_provider( $ticket_id );
+
+		return $provider->get_ticket( $event_id, $ticket_id );
+	}
+
+	/**
+	 * Create event and RSVP, return RSVP object.
+	 * Also sets timezone for event as this is needed for some tests.
+	 *
+	 * @param array $args
+	 * @return Tribe__Tickets__RSVP
+	 */
+	private function make_rsvp( $args = [] ) {
+		$event_id = $this->factory()->event->create();
+		update_post_meta( $event_id, '_EventTimezone', $this->timezone );
+		$rsvp_id  = $this->create_rsvp_ticket( $event_id, $args );
+
+		return $this->get_ticket( $event_id, $rsvp_id );
+	}
+
+	/**
+	 * Create event and Tribe Commerce Ticket, return Ticket object.
+	 * Also sets timezone for event as this is needed for some tests.
+	 *
+	 * @param integer $cost
+	 * @param array $args
+	 * @return Tribe__Tickets__Commerce__PayPal__Main
+	 */
+	private function make_ticket( $cost = 1, $args = [] ) {
+		$event_id = $this->factory()->event->create();
+		update_post_meta( $event_id, '_EventTimezone', $this->timezone );
```

Comment:
Since we do this several times, is it worth it to create a function for it and just set the timezone every time we create an event?

Something like
```
public function create_event( $args = [] ) {
	$event_id = $this->factory()->event->create( $args );
	update_post_meta( $event_id, '_EventTimezone', $this->timezone );
	return $event_id;
}```

Then in the rest of the functions/tests we can just call `$event_id = $this->create_event()` nd not worry about timezone?